core/jobs.py

```diff
@@ -139,7 +139,9 @@ def _create_new_job():
         return transaction_services.run_in_transaction(_create_new_job)
 
     @classmethod
-    def enqueue(cls, job_id, queue_name, additional_job_params=None):
+    def enqueue(
```

Comment:
Hmm, this looks like an odd pattern, @seanlip, can you advise?

Reply:
Yeah, this is indeed a bit odd. The function signature is fine but you shouldn't need the inner functions. Main code should call enqueue without supplying a test_only arg; test code should call enqueue supplying that arg.

Same thing for register_completion(). In general try to fork/branch the code as little as possible, since you want your tests to actually test the code that's running in production, rather than an isolated snippet that never runs outside the confines of the test.

Reply:
@hungphi98 can you try to go with the approach that Sean suggested?

Reply:
Thank you for your feedback. I will try to make another attempt.